src/dvclive/live.py

```diff
@@ -379,7 +391,11 @@ def log_metric(
         self.summary = set_in(self.summary, metric.summary_keys, val)
         logger.debug(f"Logged {name}: {val}")
 
-    def log_image(self, name: str, val):
+    def log_image(
+        self,
+        name: str,
+        val: Union[np.ndarray, matplotlib.figure.Figure, PIL.Image, StrPath],
```

Comment:
It looks like `StrPath` expects `Path`:

https://github.com/iterative/dvclive/blob/38f2f51c300a847e60894eff47f65fc36e4a9a25/src/dvclive/utils.py#L29

The `could_log()` method instead checks for `PurePath`, which is the [parent](https://docs.python.org/3/library/pathlib.html) of `Path`:

https://github.com/iterative/dvclive/blob/38f2f51c300a847e60894eff47f65fc36e4a9a25/src/dvclive/plots/image.py#L28

Maybe we should change `StrPath` to use `PurePath`?

Reply:
Great catch. I don't know 100% of the codebase, but if we are not using `pathlib.Path` methods to read/write files, we should be good using only `PurePath` ;)

Reply:
I also change two instance checks in `log_image` and `log_artifact`.